incubator/hnc/internal/reconcilers/object.go

```diff
@@ -764,6 +766,18 @@ func (r *ObjectReconciler) SetupWithManager(mgr ctrl.Manager, maxReconciles int)
 	target.SetGroupVersionKind(r.GVK)
 	opts := controller.Options{
 		MaxConcurrentReconciles: maxReconciles,
+
+		// Unlike the other HNC reconcilers, the object reconciler can easily be affected by objects
+		// that do not directly cause reconciliations when they're modified - see, e.g.,
+		// https://github.com/kubernetes-sigs/multi-tenancy/issues/1154. To address this, replace the
+		// default exponential backoff with one with a 5s cap.
+		//
+		// I chose five seconds since I feel like that's about how much time it would take you to check
+		// to see if something's wrong, realize it hasn't been propagated, and then try again. This
+		// should be a rare enough case that I very much doubt that it would have much material impact
+		// on HNC, unless something's causing _every_ object propagation to fail (in which case the
+		// global rate limiter will kick in to protect the cluster).
+		RateLimiter: workqueue.NewItemExponentialFailureRateLimiter(250*time.Millisecond, 5*time.Second),
```

Comment:
My only insight here is that during times of heavey load on a cluster, especially etcd, requests can take quite a while. I believe the etcd request timeout is 10 seconds (which we've been hit by a few times on our larger clusters). I also think the api-server timeout is something like 30 seconds?

So while 5 seconds is more than enough for a healthy cluster, it could cause further issues on an unhealthy cluster because we'd timeout and retry often-ish in the context of a slow cluster. 

I'm not sure it's worth changing now, but maybe something we'd come back to in the future.

Reply:
Changed to 10s